oss-cloud-backend/databaseService.js

```diff
@@ -1,97 +1,118 @@
-      if (error) {
-        reject(error);
-      }
-      resolve(data.Items.map(item => attr.unwrap(item)));
-    });
-  });
-};
-
-module.exports.checkUsername = username => {
-  var params = {
-    TableName: "contributors",
-    Key: {
-      username: username
-    }
-  };
-  return new Promise((resolve, reject) => {
-    let retval;
-    docClient.get(params, function(err, data) {
-      if (err) {
-        reject(err);
-      }
-      if (Object.keys(data).length === 0) {
-        resolve(false);
-      }
-      resolve(true);
-    });
-  });
-};
-
-// saves contributor object in database
-// params: contributor
-module.exports.addContributor = contributor => {
-  var params = {
-    TableName: "contributors",
-    Item: contributor
-  };
-  return new Promise((resolve, reject) => {
-    let retval;
-    docClient.put(params, function(err, data) {
-      if (err) {
-        reject(err);
-      } else {
-        resolve();
-      }
-    });
-  });
-};
+const dynamodb = require('serverless-dynamodb-client');
+
+const docClient = dynamodb.doc;
+const rawClient = dynamodb.raw;
+const attr = require('dynamodb-data-types').AttributeValue;
+
+module.exports.getContributor = (username) => {
+  const params = {
+    TableName: 'contributors',
+    Key: {
+      username,
+    },
+  };
+  return new Promise((resolve, reject) => {
+    docClient.get(params, (error, data) => {
+      if (error) {
+        reject(error);
+      }
+      resolve(data);
+    });
+  });
+};
+
+module.exports.updateContributorPullRequests = (username, pullRequests) => {
+  const params = {
+    TableName: 'contributors',
+    Key: {
+      username,
+    },
+    UpdateExpression: 'SET contributions = :prs',
+    ExpressionAttributeValues: {
+      ':prs': pullRequests,
+    },
+    ReturnValues: 'UPDATED_NEW',
+  };
+  return new Promise((resolve, reject) => {
+    docClient.update(params, (error) => {
+      if (error) {
+        reject(error);
+      }
+      resolve(true);
+    });
+  });
+};
+
+module.exports.getAllContributors = () => {
+  const params = {
+    TableName: 'contributors',
+  };
+  return new Promise((resolve, reject) => {
+    rawClient.scan(params, (error, data) => {
+      if (error) reject(error);
+      resolve(data.Items.map((item) => attr.unwrap(item)));
+    });
+  });
+};
+
+module.exports.checkUsername = (username) => {
+  const params = {
+    TableName: 'contributors',
+    Key: {
+      username,
+    },
+  };
+  return new Promise((resolve, reject) => {
+    let retval;
+    docClient.get(params, (err, data) => {
+      if (err) {
+        reject(err);
+      } else {
+        if (Object.keys(data).length === 0) {
+          retval = false;
+        } else retval = true;
+        resolve(retval);
+      }
+    });
+  });
+};
+
+// saves contributor object in database
+// params: contributor
+module.exports.addContributor = (contributor) => {
+  const params = {
+    TableName: 'contributors',
+    Item: contributor,

```

Comment:
I'm a little bit confused with comma added on the last property of object. There are a lot of changes where you added comma. Is that what lint suggested?

Reply:
Yes, lint suggested it. Changing that rule might not be a bad idea.